woocommerce/woocommerce-ios
# [Backend Receipts] Add backend vs local receipt eligibility usecase checks

Closes #11827


## Description
This PR introduces an eligibility check for backend receipts, so the option will be available for sites running WooCommerce 8.7.0+ (not launched yet).

We render either the legacy or the new `See receipt` based  the WooCommerce version, and if there are existing receipts already saved on device (only for paid orders): 
* If there are existing receipts on device, we just follow the existing logic and display locally-generated receipts.
* If there are no existing receipts on device, we display the new `See receipt` for eligible stores, in order to fetch one from the backend.

## Changes
- Updated `[Debug] See backend receipt` to `See Receipt`, and made them mutually exclusive: The row will either show a link to the old receipt, to the new one, or no row.
- Added an eligibility checker, we only show the new logic if: locally-generated receipt doesn't yet exist, feature flag is enabled, and WC version is supported. 

## Testing instructions
* On any store with a current WooCommerce version, confirm that the legacy flow works as expected, that is: An unpaid order shows no "See receipt" row in order details. When paying for the order via a card present method, saving the receipt, and refreshing the order (either navigate back and in again, or use pull to refresh) shows the "See receipt" row. Tapping the row, displays the old receipt (screenshot for comparison below)
* On an eligible store (you can use https://mywootestingstore.mystagingwebsite.com/): 
  * Go to an unpaid order (eg: "Pending Payment"), and observe that we do not render a `See receipt` row.
  * Go to an paid order (eg: "Completed"), and observe that `See receipt` row is rendered and a backend receipt is shown (screenshot for comparison below).

## Screenshots
| Has local receipt | Eligible for backend receipt | Awaiting payment |
|--------|--------|--------|
| ![Simulator Screenshot - iPhone 15 - 2024-02-02 at 10 13 03](https://github.com/woocommerce/woocommerce-ios/assets/3812076/857fa3b6-3c65-4389-bb25-65176b69dd98) | ![simulator_screenshot_B629D3E8-A20A-4EF9-B3A1-E4943DD0D83B](https://github.com/woocommerce/woocommerce-ios/assets/3812076/e6bb04c7-c548-4ef6-b648-6a387b623f12) | ![2024-02-02 awaiting payment](https://github.com/woocommerce/woocommerce-ios/assets/3812076/37840bf6-6174-487c-81c0-4f078a227190) |

Note that there's no summary description under `Paid` for new receipts, this is expected since currently the `paymentSummary` performs a check on payment methods to see if a card present was used. We'll attempt to improve this separately. Logged at: https://github.com/woocommerce/woocommerce-ios/issues/11878.

| Old receipt | New receipt |
|--------|--------|
| ![Simulator Screenshot - iPhone 15 - 2024-02-02 at 12 02 00](https://github.com/woocommerce/woocommerce-ios/assets/3812076/529b7259-029d-4595-b8f4-7ac17265e4b5) | ![Simulator Screenshot - iPhone 15 - 2024-02-02 at 12 01 11](https://github.com/woocommerce/woocommerce-ios/assets/3812076/c27dc5cb-41fd-4278-a775-33c09e2f5c19) |
